python/TestHarness/schedulers/RunPBS.py

```diff
@@ -70,15 +90,37 @@ def hasTimedOutOrFailed(self, job_data):
                     job.addCaveats('TESTHARNESS EXCEPTION')
                 return True
 
+    def _optimizeSelect(self, job):
+        """ Determine the optimal select statement for PBS """
+
+        slots = int(job.getMetaData().get('QUEUEING_NCPUS', 1))
+        procs_per_node = int(self._getPBSResources())
+        num_full_chunks = slots / procs_per_node
+        remaining_ranks = slots % procs_per_node
+
+        # slots > procs_per_node
+        if num_full_chunks and remaining_ranks:
+            select_ncpus = 'ncpus=%d+%d:ncpus=%d' % (remaining_ranks, num_full_chunks, procs_per_node)
+
+        # slots < procs_per_node
+        elif remaining_ranks:
+            select_ncpus = 'ncpus=%d' % (remaining_ranks)
+
+        # slots = procs_per_node
+        else:
+            select_ncpus = 'ncpus=%d' % (procs_per_node)
```

Comment:
This logic is definitely not correct. There are indeed three cases but these aren't it.

1) both num_full_chunks and remaining ranks are nonzero (slots >= procs_per_node and not divisible by procs_per_node)
2) num_full_chunks is zero (slots < procs_per_node)
3) num_remaining_ranks is zero (slots >= procs_per_node but evenly divisable)

This is absolutely not working right now.

Reply:
Hmmm... then I don't understand. This totally creates the right qsub select statement for me.

Reply:
example of a 1K core run on rod with 32 cores available per node:
```pre
#PBS -l select=1:ncpus=8+31:ncpus=32
```
8 + (31 * 32) = 1000
remaining_ranks + (num_full_chunks * procs_per_node)

I need help understanding where I am going wrong with this.